contracts/helpers/Rebalancer.sol

```diff
@@ -0,0 +1,152 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.19;
+
+import "oz/interfaces/IERC20.sol";
+import "oz/token/ERC20/utils/SafeERC20.sol";
+
+import { ITransmuter } from "interfaces/ITransmuter.sol";
+
+import { AccessControl, IAccessControlManager } from "../utils/AccessControl.sol";
+import "../utils/Constants.sol";
+import "../utils/Errors.sol";
+
+/// @title Rebalancer
+/// @author Angle Labs, Inc.
+/// @notice Contract market makers building on top of Angle can use to rebalance the reserves of the protocol
+contract Rebalancer is AccessControl {
+    using SafeERC20 for IERC20;
+
+    struct Order {
+        // Total budget allocated to subsidize the swaps between the tokens associated to the order
+        uint256 subsidyBudget;
+        // Premium paid (in `BASE_9`)
+        uint256 premium;
+    }
+
+    /// @notice Reference to the `transmuter` implementation this contract aims at rebalancing
+    ITransmuter public immutable transmuter;
+    /// @notice AgToken handled by the `transmuter` of interest
+    address public immutable agToken;
+    /// @notice Maps a `(tokenIn,tokenOut)` pair to details about the subsidy potentially provided on
+    /// `tokenIn` to `tokenOut` rebalances
+    mapping(address tokenIn => mapping(address tokenOut => Order)) public orders;
+    /// @notice Gives the total subsidy budget for each `tokenOut`
+    mapping(address tokenOut => uint256 maxBudget) public budget;
```

Comment:
This part is only for the recover function?

We don't really need to update everytime just to verify that governance won't withdraw more than it can. I would remove it

Reply:
This part is for recovery function and when you set a new order to make sure you have the balance ahead.
Like it's to ensure that every order is already prefunded.